template/Cargo.toml

```diff
@@ -21,6 +21,10 @@ test = false
 #ENDIF
 
 [dependencies]
+#IF false
+# The version which is used to insert the link to examples with a suitable tag
+esp-hal-version-full = "1.0.0"
```

Comment:
I'd say instead of this mess, let's strip the prefix in code, add enough `.0`s to make it a full semver version, and create a new template variable from that. We will always have a relevant tag for a version we release, even if we simplify out the patch version here.